meal-sharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,136 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+router.get("/", async (request, response) => {
+  try {
+    let meal = await knex("meal");
+
+    if ("maxprice" in request.query) {
+      mealMaxPrice = meal.filter(
+        (meal) => Number(meal.price) < Number(request.query.maxprice)
+      );
+      if (isNaN(request.query.maxprice)) {
```

Comment:
It is better to check if the query parameter is not valid before querying the database.
That way you avoid an unnecessary query and thus, make your application respond faster.
This is a general idea to have in mind: You should try to keep your queries to the db to a minimum, cause in this case the meal app contains only a few records, but what if it grows to millions and millions? It will start to get slower..